File: desktopclient/src/main/java/com/goexp/galgame/gui/db/mongo/GameDB.java
package com.goexp.galgame.gui.db.mongo;

import com.goexp.common.db.mongo.DBOperatorTemplate;
import com.goexp.galgame.gui.model.Game;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.List;

import static com.mongodb.client.model.Filters.eq;
import static com.mongodb.client.model.Updates.set;

public class GameDB {

    private static DBOperatorTemplate tlp = new DBOperatorTemplate("galgame", "game");

    public static class StarDB {

        private final Logger logger = LoggerFactory.getLogger(StarDB.class);


        public void update(Game game) {

            logger.debug("<update> {}", game);

            tlp.exec(documentMongoCollection -> {
                documentMongoCollection.updateOne(eq("_id", game.id), set("star", game.star));
            });
        }

//        public void batchUpdate(List<Game> games) {
//            try {
//
//
//                var param = new Object[games.size()][];
//
//                for (var i = 0; i < games.size(); i++) {
//                    var img = games.get(i);
//                    param[i] = new Object[]{img.star, img.id};
//                }
//                runner.batch("update game_star set `star`=? where id=?", param);
//
//
//            } catch (SQLException e) {
//                e.printStackTrace();
//            }
//        }

    }

    public static class StateDB {


        public void update(Game game) {
            tlp.exec(documentMongoCollection -> {
                documentMongoCollection.updateOne(
                        eq("_id", game.id)
                        , set("state", game.state.get().getValue())
                );
            });
        }


        public void batchUpdate(List<Game> games) {


            tlp.exec(documentMongoCollection -> {

                games.forEach(game -> {
                    documentMongoCollection.updateOne(
                            eq("_id", game.id)
                            , set("state", game.state.get().getValue())
                    );
                });

            });

        }
    }
}

File: desktopclient/src/main/scala/com/goexp/galgame/gui/db/mongo/GameDB.scala
package com.goexp.galgame.gui.db.mongo

import java.util

import com.goexp.common.db.mongo.DBOperatorTemplate
import com.goexp.galgame.common.model.GameState
import com.goexp.galgame.gui.model.Game
import com.mongodb.client.model.Filters.{eq => equal}
import com.mongodb.client.model.Updates.set
import org.slf4j.LoggerFactory
import com.goexp.galgame.common.db.mongo.DB_NAME


object GameDB {
  private val tlp = new DBOperatorTemplate(DB_NAME, "game")

  object StarDB {
    private val logger = LoggerFactory.getLogger(StarDB.getClass)

    def update(game: Game): Unit = {
      logger.debug("<update> {}", game)
      tlp.exec(documentMongoCollection => {
        documentMongoCollection.updateOne(equal(game.id), set("star", game.star))
      })
    }
  }

  object StateDB {
    def update(game: Game): Unit =
      tlp.exec(documentMongoCollection => {
        documentMongoCollection.updateOne(equal(game.id), set("state", game.state.get.getValue))
      })

    def update(brandId: Int): Unit =
      tlp.exec(documentMongoCollection => {
        documentMongoCollection.updateMany(equal("brandId", brandId), set("state", GameState.BLOCK.getValue))

      })

    def batchUpdate(games: util.List[Game]): Unit =
      tlp.exec(documentMongoCollection => {
        games.forEach((game: Game) => {
          documentMongoCollection.updateOne(equal(game.id), set("state", game.state.get.getValue))
        })
      })
  }

}